react/player/src/__tests__/player.test.tsx

```diff
@@ -0,0 +1,61 @@
+import { useSubscribedState } from "@player-ui/react-subscribe";
+import { act, render, screen } from "@testing-library/react";
+import React, { Suspense, type ComponentType } from "react";
+import { describe, expect, test } from "vitest";
+import { ReactPlayer, type ReactPlayerPlugin } from "../player";
+
+describe("ReactPlayer", () => {
+  test("switching from suspending component to an alternative component does not hang", async () => {
+    let rp: ReactPlayer;
+
+    const viewUpdateEvent = { type: "alternative-component", id: "1" };
+
+    const TestPlugin: ReactPlayerPlugin = {
+      name: "test-plugin",
+      applyReact: (reactPlayer) => {
+        rp = reactPlayer;
+
+        // Register a dummy asset for the alternative component
+        // This prevents ActualPlayerComp from crashing if it renders before being unmounted
+        const dummyAsset = { type: viewUpdateEvent.type };
```

Comment:
Could we do another test and have this be `type: 'generic'` or something, then render the base component followed by the alternative view to very suspense works between them?

Other than that everything looks 💯 . Thank you!